src/storage/StorageModule.ts

```diff
@@ -1,32 +1,292 @@
+        return hexlify(this.latestJoinChannelBlockHash);
+    }
+
+    /**
+     * Store the block hash for an exit channel block
+     */
+    storeExitChannelBlockHash(
+        blockHash: string,
+        exitChannelBlock: ExitChannelBlockStruct
+    ) {
+        this.exitChannelBlockMap.set(blockHash, exitChannelBlock);
+        this.latestExitChannelBlockHash = blockHash;
+    }
+
+    /**
+     * Retrieve the block hash for an exit channel block
+     */
+    getExitChannelBlock(blockHash: string): ExitChannelBlockStruct | undefined {
+        return this.exitChannelBlockMap.get(blockHash);
+    }
+
     getLatestExitChannelBlockHash(): string {
-        // TODO
-        return ethers.ZeroHash;
+        return hexlify(this.latestExitChannelBlockHash);
+    }
+    //#endregion
+
+    //#region Total deposits and withdrawals management methods
+
+    /**
+     * Add to the total deposits
+     * Doesn't handle data yet, just the amount
+     */
+    private addToTotalDeposits(balance: BalanceStruct) {
+        //TODO: create a method to add two balances together
+        this.totalDeposits.amount += toBeHex(balance.amount);
+        console.log(`Total deposits updated: ${this.totalDeposits}`);
     }
 
+    /**
+     * Get the total deposits
+     */
     getTotalDeposits(): { amount: BigNumberish; data: string } {
-        // TODO
-        return { amount: 0, data: "0x" };
+        return {
+            amount: this.totalDeposits.amount,
+            data: hexlify(this.totalDeposits.data)
+        };
+    }
+
+    /**
+     * Add to the total withdrawals
+     * Doesnt handle data yet, just the amount
+     */
+    addToTotalWithdrawals(balance: BalanceStruct) {
+        this.totalWithdrawals.amount += toBeHex(balance.amount);
+        console.log(`Total withdrawals updated: ${this.totalWithdrawals}`);
     }
 
+    /**
+     * Get the total withdrawals
+     */
     getTotalWithdrawals(): { amount: BigNumberish; data: string } {
-        // TODO
-        return { amount: 0, data: "0x" };
+        return {
+            amount: this.totalWithdrawals.amount,
+            data: hexlify(this.totalWithdrawals.data)
+        };
     }
+    //#endregion
 
-    getPreviousBlockHash(
-        forkCnt: number,
-        transactionCnt: number
-    ): string | undefined {
-        // TODO
-        return ethers.ZeroHash;
+    //#region Helper methods
+
+    private makeKey(forkCnt: number, blockHeight: number): string {
+        return `${forkCnt}-${blockHeight}`;
+    }
+
+    private parseKey(key: string): [number | undefined, number | undefined] {
+        const parts = key.split("-");
+        if (parts.length !== 2) {
+            return [undefined, undefined];
+        }
+
+        const forkCnt = parseInt(parts[0], 10);
+        const blockHeight = parseInt(parts[1], 10);
+
+        if (isNaN(forkCnt) || isNaN(blockHeight)) {
+            return [undefined, undefined];
+        }
+
+        return [forkCnt, blockHeight];
     }
+    //#endregion
```

Comment:
best to partition this file early - will be easier for you to work in isolation also